projects/fw_102/src/main.c

```diff
@@ -0,0 +1,38 @@
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#define COUNTER_DURATION 500
+
+typedef struct {
+  uint8_t counter_a;
+  uint8_t counter_b;
+  int turn;
```

Comment:
You don't need to have this variable in the struct here. Since you are incrementing counter_a on every timer callback, you can use it's value (the exact same way as you are using 'turn') to determine whether or not to increment counter_b